app/tests/retina_api_tests/test_views.py

```diff
@@ -143,6 +145,49 @@ def test_with_data(
             assert response.status_code == status.HTTP_200_OK
             assert response.content == b'{"directories":[],"images":[]}'
 
+    def test_only_load_metaio_images(
+        self, client, archive_patient_study_image_set
+    ):
+        cache.clear()
+        user = get_user_from_str("retina_user")
+        archive_patient_study_image_set.archive1.add_user(user)
+        pk = archive_patient_study_image_set.study113.pk
+
+        for (index, image) in enumerate(
+            archive_patient_study_image_set.images111
+        ):
+            if index % 2 == 0:
+                continue
+            for image_file in image.files.all():
+                image_file.image_type = ImageFile.IMAGE_TYPE_DZI
+                image_file.save()
+
+        response = self.perform_request_as_user(client, user, pk)
+        assert response.status_code == status.HTTP_200_OK
+        result = json.loads(response.content)
+        assert len(result["images"]) == len(
+            archive_patient_study_image_set.images113
+        )
+        res_img_ids = {i["id"] for i in result["images"]}
+        exp_img_ids = {
+            str(i.pk) for i in archive_patient_study_image_set.images113
+        }
+        assert res_img_ids == exp_img_ids
+
+    def test_number_of_queries(
+        self,
+        client,
+        archive_patient_study_image_set,
+        django_assert_num_queries,
+    ):
+        cache.clear()
+        user = get_user_from_str("retina_user")
+        archive_patient_study_image_set.archive1.add_user(user)
+        pk = archive_patient_study_image_set.study113.pk
+
+        with django_assert_num_queries(22):
```

Comment:
Would prefer to use `max_num_queries` instead as having the exact number can be quite fragile.